Mathlib/RingTheory/TensorProduct/IncludeLeftSubRight.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2025 Yong-Gyu Choi. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yong-Gyu Choi
+-/
+module
+
+public import Mathlib.Algebra.Category.Ring.Constructions
+public import Mathlib.RingTheory.Flat.FaithfullyFlat.Basic
+
+/-!
+# Exactness properties of the difference map for tensor products
+
+For an `R`-algebra `S`, we collect some properties of the `R`-linear map `S →ₗ[R] S ⊗[R] S` given
+by `s ↦ (s ⊗ₜ[R] 1) - (1 ⊗ₜ[R] s)`.
+
+## Main definitions
+
+* `includeLeftSubRight`: The `R`-linear map sending `s : S` to `s ⊗ₜ[R] 1 - 1 ⊗ₜ[R] s`.
+* `ExactIncludeLeftSubRight`: Exactness of the sequence `R → S → S ⊗[R] S` with the right map given
+  by `includeLeftSubRight`
+* `toEqLocusOfInclusion`: The ring map from `R` to the equalizer locus in `S` of the two maps
+  `s ↦ s ⊗ₜ (1 : S)` and `s ↦ (1 : S) ⊗ₜ s`.
+
+## Main results
+
+* `exactIncludeLeftSubRight_of_faithfullyFlat`: `ExactIncludeLeftSubRight R S` is true for any
+  faithfully flat `R`-algebras `S`.
+* `toEqLocusOfInclusion_surjective`: /-- If `ExactIncludeLeftSubRight R S` is true, then
+  `toEqLocusOfInclusion R S` is surjective. -/
+
+-/
+
+@[expose] public section
+
+open scoped TensorProduct
+
+namespace Algebra.TensorProduct
+
+universe uR uS uT
+
+section IncludeLeftSubRight
+
+variable {R : Type uR} [CommSemiring R]
+variable {S : Type uS} [Ring S] [Algebra R S]
+
+variable (R S) in
+/-- The `R`-linear map `S →ₗ[R] S ⊗[R] S` sending `s : S` to `(s ⊗ₜ[R] 1) - (1 ⊗ₜ[R] s)`. -/
+def includeLeftSubRight : S →ₗ[R] S ⊗[R] S :=
+  (includeLeft (R := R) (S := R) (A := S) (B := S)).toLinearMap -
+    (includeRight (R := R) (A:= S) (B := S)).toLinearMap
+
+@[simp]
+lemma includeLeftSubRight_apply (s : S) : includeLeftSubRight R S s = s ⊗ₜ[R] 1 - 1 ⊗ₜ[R] s :=
+  rfl
+
+/-- `includeLeftSubRight R S` vanishes in the range of `algebraMap R S`. -/
+lemma includeLeftSubRight_zero_of_mem_range {s : S} (hs : s ∈ Set.range ⇑(algebraMap R S)) :
+    (includeLeftSubRight R S) s = 0 := by
+  obtain ⟨_, hr⟩ := Set.mem_range.mp hs
+  simp [includeLeftSubRight, ← hr]
+
+/-- `includeLeftSubRight` is compatible with `distribBaseChange` and `lTensor`. -/
+lemma includeLeftSubRight_distribBaseChange (T : Type uT) [CommRing T] [Algebra R T] :
+    (includeLeftSubRight T (T ⊗[R] S)).restrictScalars R =
+    ((TensorProduct.AlgebraTensorModule.distribBaseChange R T S S).restrictScalars R).toLinearMap ∘ₗ
+      ((includeLeftSubRight R S).lTensor T) := by
+  simp only [includeLeftSubRight, LinearMap.lTensor_sub, LinearMap.comp_sub,
+    distribBaseChange_includeLeft_lTensor, distribBaseChange_includeRight_lTensor]
+  rfl
+
+variable (R S) in
+/-- For an `R`-algebra `S`, this asserts that the maps `algebraMap : R → S` and
+`includeLeftSubRight R S : S → S ⊗[R] S` form an exact pair. -/
+def ExactIncludeLeftSubRight : Prop :=
+  Function.Exact ⇑(algebraMap R S) ⇑(includeLeftSubRight R S)
```

Comment:
```suggestion
  Function.Exact (algebraMap R S) (includeLeftSubRight R S)
```
And I am guessing replacing `algebraMap R S` with `Algebra.linearMap R S` would make it easier to use.